internal/app/pactproxy/interaction.go

```diff
@@ -104,6 +110,30 @@ func LoadInteraction(data []byte, alias string) (*interaction, error) {
 	return interaction, nil
 }
 
+func isJSONRequest(request map[string]interface{}) (bool, error) {
+	headers, hasHeaders := request["headers"]
+	if !hasHeaders {
+		return false, nil
+	}
+
+	parsed, ok := headers.(map[string]string)
```

Comment:
http headers are normally considered `map[string][]string` don't know why but here's the [link](https://pkg.go.dev/net/http#Header)